Server.js

```diff
@@ -1,12 +1,26 @@
 const express = require('express');
 const app = express();
-const port = 3000;
+const port = process.env.PORT || 3000;
 
 // define the ping route
 app.get("/ping",(req,res)=>{
     res.send("pong")
 })
 
+app.all("*", (req,res,next) => {
+  let err = new Error()
+  err.status = `Can't find ${req.originalUrl} on the server`
+  err.statusCode = 404;
+
+  next(err);
+})
+// Error handling middleware for undefined webpages
+app.use((error, req, res, next) => {
+  
+  res.status(500).send(error);
```

Comment:
Consider adding a more detailed error message for the 500 status code. This can help in debugging and understanding the nature of the error. [medium]